nixos/modules/system/boot/systemd.nix

```diff
@@ -907,6 +911,13 @@ in
             type = service.serviceConfig.Type or "";
             restart = service.serviceConfig.Restart or "no";
             hasDeprecated = builtins.hasAttr "StartLimitInterval" service.serviceConfig;
+            hasStartCmd = svc:
+              svc.script != ""
+              || svc.serviceConfig?ExecStart
+              || svc.serviceConfig?ExecStop
+              || svc.serviceConfig?SuccessAction;
```

Comment:
```suggestion
```

... because IMHO a `SuccessAction` (or a `FailureAction`) only make sense when a service has succeeded but if there's only a `SuccessAction` and no `Exec{Start,Pre}` will `SuccessAction` ever fire?

Reply:
I'm not sure! I am just following the error message `systemd-analyze verify` gives:
```ShellSession
$ cat test.service
[Unit]
Description = example

$ systemd-analyze verify test.service 
/nix/store/sbd3spyaymdgvif20irx4ijrs1smzvf9-unit-nginx.service/nginx.service:54: Unknown system call group, ignoring: @mincore
test.service: Service has no ExecStart=, ExecStop=, or SuccessAction=. Refusing.
Unit test.service has a bad unit file setting.
```

Reply:
`SuccessAction=` is listed under `man systemd.unit`, though, so perhaps this is incorrect for services.

Also I would really preferred to have made this a type error, but it's so complicated 😮‍💨 .

Reply:
Oh, @jsoo1 then you should run with that. In the future, you could provide justification in the PR / commit description.